file.go

```diff
@@ -1,37 +1,127 @@
 package jcsv
 
-import "os"
+
+import (
+	"encoding/csv"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+)
 
 type file struct {
-	// TODO: define variables if needed
+	jsonData []uint8
+	csvData [][]string
+	csvHeader []string
 }
 
 func ParseJsonFile(path string) (file, error) {
-	// TODO: open and read the given file into your `file` object
-	return file{}, nil
+	var fileObj file
+	fileStream,fileOpenError := os.Open(path)
+	if fileOpenError!=nil{
+		return file{},fileOpenError
+	}
+	var fileReadError error
+	fileObj.jsonData,fileReadError=ioutil.ReadAll(fileStream)
+	fileStream.Close()
+	if fileReadError!=nil {
+		return file{},fileReadError
+	}
+	return fileObj,nil
 }
 
 func ParseOpenedJsonFile(f *os.File) (file, error) {
-	// TODO: read the given file into your `file` object
-	return file{}, nil
+	var fileRead file
+	var fileReadError error
+	fileRead.jsonData,fileReadError=ioutil.ReadAll(f)
+	if fileReadError!=nil {
+		return file{},fileReadError
+	}
+	return fileRead,nil
 }
 
 func ParseCsvFile(path string, hasHeaders bool) (file, error) {
-	// TODO: open and read the given file into your `file` object
-	return file{}, nil
+	var returnFile file
+	csvFile, err := os.Open(path)
+	if err != nil {
+		fmt.Println(err)
+		return file{},err
+	}
+	readBuffer,readError := csv.NewReader(csvFile).ReadAll()
+	if readError!=nil{
+		return file{},readError
+	}
+	var i int
+	if hasHeaders {
+		for j := 0; j < len(readBuffer[0]); j++ {
+			returnFile.csvHeader = append(returnFile.csvHeader, readBuffer[0][j])
+		}
+		i=1
+	}else {
+		for j:=0;j<len(readBuffer[0]);j++{
+			returnFile.csvHeader=append(returnFile.csvHeader,"key" + strconv.Itoa(j))
+		}
+		i=0
+	}
+	for ;i<len(readBuffer);i++{
+		returnFile.csvData = append(returnFile.csvData, readBuffer[i])
+	}
+	csvFile.Close()
+	return  returnFile,nil
 }
 
-func ParseCsvJsonFile(f *os.File, hasHeaders bool) (file, error) {
-	// TODO: read the given file into your `file` object
-	return file{}, nil
+func ParseOpenedCsvFile(f *os.File, hasHeaders bool) (file, error) {
+	var fileObj file
+	readBuffer,readError := csv.NewReader(f).ReadAll()
+	if readError!=nil{
+		return file{},readError
+	}
+	var i int
+	if hasHeaders {
+		for j := 0; j < len(readBuffer[0]); j++ {
+			fileObj.csvHeader = append(fileObj.csvHeader, readBuffer[0][j])
+		}
+		i=1
+	}else {
+		for j:=0;j<len(readBuffer[0]);j++{
+			fileObj.csvHeader=append(fileObj.csvHeader,"key" + strconv.Itoa(j))
+		}
+		i=0
+	}
+	for ;i<len(readBuffer);i++{
+		fileObj.csvData = append(fileObj.csvData, readBuffer[i])
+	}
+	return  fileObj,nil
 }
 
 func (f file) Csv(addHeaders bool) []byte {
-	// TODO: return the file data in CSV format
-	return nil
+	var dataToBeReturned []byte
+	if addHeaders == true{
+		for j:=0;j<len(f.csvHeader);j++{
```

Comment:
If the input file was a JSON file, this will fail.

Reply:
Yes, I didn't design it that way. After reading "return the file data in CSV format", I thought it would return data,read from CSV file, in form of [ ] byte. Now, I've edited it, and it will convert data, read from Json file, to CSV.